source/distributed/fully_distributed_tria.cc

```diff
@@ -301,8 +313,140 @@ namespace parallel
     void
     Triangulation<dim, spacedim>::execute_coarsening_and_refinement()
     {
-      Assert(false, ExcNotImplemented());
-    }
+      // 1) preparation
+      {
+        // reset the coarsening/refinement flags of non-local cells
+        for (auto cell : this->cell_iterators())
+          if (cell->active() && !cell->is_locally_owned())
+            {
+              cell->clear_refine_flag();
+              cell->clear_coarsen_flag();
+
+              // start assumption:
+              if (cell->level() > 0)
+                cell->set_coarsen_flag();
+            }
+
+        // update refinement/coarsening flags of non-local cells via
```

Comment:
I think you need to call prepare_coarsening_and_refinement first, before you send this information.